bundles/go/go_fmt.moon

```diff
@@ -0,0 +1,58 @@
+import app, config, mode, signal from howl
+import Process from howl.io
+
+append = table.insert
+
+run_command = (file) ->
+  args = {}
+  for arg in config.go_fmt_command\gmatch '%S+'
+    append args, arg
+  append args, file
+  success, out, err, p = pcall Process.execute, args
+  unless success
+    log.error out
+    return
+  unless p.successful
+    log.error err
+    return
+
+reload_buffer = (buffer) ->
```

Comment:
The `@file` setter doesn't have to change, just the implementation of  `buffer\reload`. E.g. it could just do what is being done here, if  `{preserve_undo: true}` is passed in. This seems like a case where improving the underlying API will have future benefit.

I also wonder if undo preserving reload is should be the default, but that requires a wider conversation.